### PR TITLE
replace print statements with logger

### LINKED ISSUE #1373: No more prints
There are still occasions where `print` has remained and should be replaced by the `logger.*`. Some can remain, e.g. the notification about daemonizing or missing dependencies. This issue will be updated as I find more locations.

- [ ] `src/api.py`

### PR DESCRIPTION
**Pull request by navjotcis**
I have replaced print statement with logger for below given module
`network/randomtrackingdict.py`
`singleinstance.py`

### CODE REVIEW

**Review by PeterSurda — CHANGES_REQUESTED** [review]
- the test in randomtrackingdict should be refactored as a unit test
- remove singleinstnace from PR
- filesystem is OK

**Review by PeterSurda — CHANGES_REQUESTED** [review]
Move tests into unit tests.

**Comment by PeterSurda** [comment]
- remove main section from `randomtrackingdict.py`
- fix errors reported by travis

File: src/tests/test_randomtrackingdict.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            try:
+                k.randomKeys(100)
+                print('bad')
+            except KeyError:
+                pass
+            for i in retval:
+                del k[i]
+        a.append(time())
+
+        for x in range(len(a) - 1):
```
Comment (PeterSurda) [inline src/tests/test_randomtrackingdict.py:46]: This is debugging information. We should check if this is not too slow. Let's say all values of `a[x + 1] - a[x]` need to be less than `10.0`.

**Review by PeterSurda — CHANGES_REQUESTED** [review]
Please do the requested changes and fix code quality.

**Comment by PeterSurda** [comment]
Also you can't write "Fixes #xxxxx" unless it's a complete fix, that screws up the dependency tracking.

**Comment by navjotcis** [comment]
I have made the suggested changes not it is ready to merge.

File: src/tests/test_randomtrackingdict.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    @staticmethod
+    def randString():
+        """helper function for tests, generates a random string"""
+        retval = b''
+        for _ in range(32):
+            retval += chr(random.randint(0, 255))
+        return retval
+
+    def test_check_randomtrackingdict(self):
+        """Check the logic of RandomTrackingDict class"""
```
Comment (PeterSurda) [inline src/tests/test_randomtrackingdict.py:23]: `"""Test performance of RandomTrackingDict"""`

File: src/tests/test_randomtrackingdict.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                k.randomKeys(100)
+                self.fail("bad")
+            except KeyError:
+                pass
+            for i in retval:
+                del k[i]
+        a.append(time())
+
+        for x in range(len(a) - 1):
+            self.assertLess(a[x + 1] - a[x], 10)
```
Comment (PeterSurda) [inline src/tests/test_randomtrackingdict.py:47]: Maybe this needs to be `10.0`, I'm not fully sure, we'll see once the import is fixed.

**Review by PeterSurda — CHANGES_REQUESTED** [review]
Please fix import.

File: src/network/__init__.py [hunk 4]
```diff
@@ -8,6 +8,7 @@
 from downloadthread import DownloadThread
 from invthread import InvThread
 from networkthread import BMNetworkThread
+from .randomtrackingdict import RandomTrackingDict
```
Comment (PeterSurda) [inline src/network/__init__.py:11]: why dot at the beginning?

**Review by PeterSurda — CHANGES_REQUESTED** [review]
fix

File: src/network/__init__.py (hunk 4 above)
Comment (navjotcis) [reply src/network/__init__.py]: checking with every possible way

**Pull request closed without merging**